### PR TITLE
[test]recommendation-wide-n-deep/sentiment analysis

### PR DESCRIPTION
**Pull request by xiaxin1998**

### CODE REVIEW

**Comment by xiaxin1998** [comment]
http://10.239.47.211:18888/view/ZOO-PR/job/ZOO-PR-Pip-Apps-Integration/139/ Jekin passed @hkvision

File: apps/run-app-tests-pip.sh [hunk 1]
```diff
@@ -451,5 +451,57 @@ now=$(date "+%s")
...
+echo "#4 start app test for wide_n_deep"
+start=$(date "+%s")
+
+# Conversion to py file and data preparation
+${ANALYTICS_ZOO_HOME}/apps/ipynb2py.sh ${ANALYTICS_ZOO_HOME}/apps/recommendation-wide-n-deep/wide_n_deep
+sed "s/end_trigger=MaxEpoch(10)/end_trigger=MaxEpoch(5)/g" ${ANALYTICS_ZOO_HOME}/apps/recommendation-wide-n-deep/wide_n_deep.py >${ANALYTICS_ZOO_HOME}/apps/recommendation-wide-n-deep/tmp_test.py
+
+# Run the example
+export SPARK_DRIVER_MEMORY=22g
+python ${ANALYTICS_ZOO_HOME}/apps/recommendation-wide-n-deep/wide_n_deep.py
```
Comment (hkvision) [inline apps/run-app-tests-pip.sh]: why aren't you running `tmp_test.py`?

File: apps/run-app-tests-pip.sh [hunk 2]
```diff
@@ -451,5 +451,57 @@ now=$(date "+%s")
...
+if [ $exit_status -ne 0 ];
+then
+    clear_up
+    echo "using_variational_autoencoder_and_deep_feature_loss_to_generate_faces failed"
```
Comment (hkvision) [inline apps/run-app-tests-pip.sh]: wrong name

File: apps/run-app-tests-pip.sh [hunk 3]
```diff
@@ -451,5 +451,57 @@ now=$(date "+%s")
...
+   echo "Finished"
+fi
+
+# Run the example
+export SPARK_DRIVER_MEMORY=12g
+python ${ANALYTICS_ZOO_HOME}/apps/sentiment-analysis/sentiment.py
+if [ $exit_status -ne 0 ];
+then
+    clear_up
+    echo "using_variational_autoencoder_and_deep_feature_loss_to_generate_faces failed"
```
Comment (hkvision) [inline apps/run-app-tests-pip.sh]: wrong name

**Comment by hkvision** [comment]
The jenkins you run is on PR #899 Please run this PR.

**Comment by hkvision** [comment]
LGTM. Can merge after jenkins pass

**Comment by hkvision** [comment]
What's the status of this PR for running jenkins?

**Comment by hkvision** [comment]
Any update?

**Comment by xiaxin1998** [comment]
http://10.239.47.211:18888/view/ZOO-PR/job/ZOO-PR-Pip-Apps-Integration/143/console   Jenkins failed and I don’t know how to fix it.

**Comment by hkvision** [comment]
Use 0.4.0.dev0 instead of 0.3.0.dev0

**Comment by hkvision** [comment]
Let's continue to work on this PR.

File: apps/recommendation-wide-n-deep/wide_n_deep.ipynb [hunk 4]
```diff
@@ -319,7 +319,7 @@
    ],
    "source": [
     "# Create an Optimizer\n",
-    "batch_size = 8000\n",
+    "batch_size = 8008\n",
```
Comment (hkvision) [inline apps/recommendation-wide-n-deep/wide_n_deep.ipynb]: don't change this file

File: apps/run-app-tests-pip.sh [hunk 5]
```diff
@@ -456,6 +511,3 @@ unset SPARK_DRIVER_MEMORY
 now=$(date "+%s")
 time1=$((now-start))
 echo "recommendation-ncf time used:$time1 seconds"
-
-# This should be done at the very end after all tests finish.
-clear_up
```
Comment (hkvision) [inline apps/run-app-tests-pip.sh]: why are you deleting these lines?

**Comment by hkvision** [comment]
1. Don't make changes to the ipynb files.
2. After passing jenkins, put your newly added scripts in the correct position.

File: apps/run-app-tests-pip.sh [hunk 6]
```diff
@@ -11,6 +11,62 @@ clear_up () {
...
+
+exit_status=$?
+if [ $exit_status -ne 0 ];
+then
+    clear_up
+    echo "recommendation-wide-n-deep failed"
+    exit $exit_status
+fi
+ unset SPARK_DRIVER_MEMORY
+ now=$(date "+%s")
```
Comment (hkvision) [inline apps/run-app-tests-pip.sh]: remove redundant space for line 32-33

File: apps/run-app-tests-pip.sh [hunk 7]
```diff
@@ -11,6 +11,62 @@ clear_up () {
...
+time4=$((now-start))
+echo "recommendation-wide-n-deep time used:$time4 seconds"
+rm ${ANALYTICS_ZOO_HOME}/apps/recommendation-wide-n-deep/tmp_test.py
+
+echo "#11 start app test for sentiment-analysis"
```
Comment (hkvision) [inline apps/run-app-tests-pip.sh]: seems there's already # 11, change to # 8?

File: apps/run-app-tests-pip.sh [hunk 8]
```diff
@@ -11,6 +11,62 @@ clear_up () {
...
+
+exit_status=$?
+if [ $exit_status -ne 0 ];
+then
+    clear_up
+    echo "sentiment-analysis failed"
+    exit $exit_status
+fi
+ unset SPARK_DRIVER_MEMORY
+ now=$(date "+%s")
```
Comment (hkvision) [inline apps/run-app-tests-pip.sh]: remove space

**Comment by xiaxin1998** [comment]
http://10.239.47.211:18888/view/ZOO-PR/job/ZOO-PR-Pip-Apps-Integration/180/ Jenkins passed

**Pull request merged**